src/YesSql.Abstractions/ConcurrencyException.cs

```diff
@@ -1,11 +1,27 @@
 using System;
+using System.Reflection.Metadata;
 
 namespace YesSql
 {
     public class ConcurrencyException : Exception
     {
-        public ConcurrencyException() : base("The document could not be updated as it has been changed by another process.")
+        private readonly string _message;
+
+        public Document Document { get; }
+
+        public ConcurrencyException(Document document)
         {
+            Document = document;
+
+            _message= $"""
+The document could not be updated as it has been changed by another process.
+     ID: {Document.Id}
+   Type: {Document.Type}
+Version: {Document.Version}
+Content: {Document.Content}        
+""";
         }
```

Comment:
I formatted the error messages and wrote them directly to the logs, which shouldn't need to be processed by an external application